### PR TITLE
[RFC] New option: g:signify_skip_filename_pattern

### PR DESCRIPTION
**Pull request by mhinz**
CC @jamessan

### CODE REVIEW

File: autoload/sy.vim [hunk 1]
```diff
@@ -147,3 +142,31 @@ endfunction
...
+    elseif has_key(g:signify_skip_filetype, 'help') && (&buftype == 'help')
+      return 1
+    endif
+  endif
+
+  if exists('g:signify_skip_filename') && has_key(g:signify_skip_filename, a:path)
+    return 1
+  endif
+
+  if exists('g:startify_skip_filename_pattern')
```
Comment (jamessan) [inline autoload/sy.vim]: `g:startify_skip_filename_pattern` -- wrong project. :) Same with the commit message.

File: autoload/sy.vim [hunk 2]
```diff
@@ -147,3 +142,31 @@ endfunction
...
+    for pattern in g:startify_skip_filename_pattern
+      if match(a:path, pattern) != -1
```
Comment (jamessan) [inline autoload/sy.vim]: Why not use `=~`?

    if a:path =~ pattern
        return 1
    endif

File: doc/signify.txt [hunk 3]
```diff
@@ -200,17 +201,24 @@ This loads Sy, but it won't look for changes. You can toggle it anytime via
...
+Don't activate the plugin for these filetypes and/or filenames. Filenames have
+to be absolute paths.
 
-NOTE: Filenames have to be absolute paths.
+These options must be |Dict|s for faster lookup.
+>
+    let g:signify_skip_filename_pattern = [ 'foo.*bar', 'tmp' ]
+<
+Don't activate the plugin for filenames matching these patterns. Internally
+this uses |match()|.
```
Comment (jamessan) [inline doc/signify.txt]: Not sure why it's worth calling out the use of `match()` here.  It's a pattern match.  People using this option should know how pattern matches work.  `match()` doesn't behave any differently in that regard.

File: autoload/sy.vim (hunk 1 above)
Comment (mhinz) [inline autoload/sy.vim]: Hahahaha. Oh my, thanks for catching that!

File: autoload/sy.vim (hunk 2 above)
Comment (mhinz) [inline autoload/sy.vim]: :+1:

File: doc/signify.txt (hunk 3 above)
Comment (mhinz) [inline doc/signify.txt]: :+1:

**Comment by mhinz** [comment]
Thanks for comments!

**Pull request merged**